src/Core/src/Platform/iOS/UIWindowExtensions.cs

```diff
@@ -0,0 +1,48 @@
+using System;
+using ObjCRuntime;
+using UIKit;
+
+namespace Microsoft.Maui.Platform
+{
+	public static class UIWindowExtensions
+	{
+		// TODO Add public docs
+		public static IWindow? GetWindow(this UIWindow? platformWindow)
+		{
+			if (platformWindow is null)
+				return null;
+
+			foreach (var window in IPlatformApplication.Current?.Application?.Windows ?? Array.Empty<IWindow>())
+			{
+				if (window?.Handler?.PlatformView == platformWindow)
+					return window;
+			}
+
+			return null;
+		}
+
+		// TODO Add public docs
+		public static IWindow? GetWindow(this UIWindowScene? windowScene)
+		{
+			if (windowScene is null)
+				return null;
+
+#pragma warning disable CA1416 // TODO: 'UIApplication.Windows' is unsupported on: 'ios' 15.0 and later
+			foreach (var window in windowScene.Windows)
+			{
+				var managedWindow = window.GetWindow();
+
+				if (managedWindow is not null)
+					return managedWindow;
+			}
```

Comment:
This should be wrapped in a `if (OperatingSystem.IsIOSVersionAtLeast(15) || OperatingSystem.IsMacCatalyst())`

Reply:
I see it was like this in the original code, so we can merge for now and review later.